yt_dlp/extractor/safari.py

```diff
@@ -117,39 +102,42 @@ class SafariIE(SafariBaseIE):
     _UICONF_ID = '29375172'
 
     def _real_extract(self, url):
+        query = {}
         mobj = self._match_valid_url(url)
 
         reference_id = mobj.group('reference_id')
         if reference_id:
-            video_id = reference_id
-            partner_id = self._PARTNER_ID
-            ui_id = self._UICONF_ID
+            query['flashvars[referenceId]'] = video_id = reference_id
+            query['wid'] = '_%s' % self._PARTNER_ID
+            query['uiconf_id'] = self._UICONF_ID
         else:
             video_id = '%s-%s' % (mobj.group('course_id'), mobj.group('part'))
 
             webpage, urlh = self._download_webpage_handle(url, video_id)
 
             mobj = re.match(self._VALID_URL, urlh.geturl())
             reference_id = mobj.group('reference_id')
-            if not reference_id:
-                reference_id = self._search_regex(
-                    r'data-reference-id=(["\'])(?P<id>(?:(?!\1).)+)\1',
-                    webpage, 'kaltura reference id', group='id')
-            partner_id = self._search_regex(
+            if reference_id:
+                query['flashvars[referenceId]'] = reference_id
+            else:
+                try:
+                    query['flashvars[referenceId]'] = self._search_regex(
+                        r'data-reference-id=(["\'])(?P<id>(?:(?!\1).)+)\1',
+                        webpage, 'kaltura reference id', group='id')
+                except RegexNotFoundError as e:
```

Comment:
Non fatal search failure still encourages opening issue:

https://github.com/yt-dlp/yt-dlp/blob/b57c15302304ab2f464a8f4dc37389c60ba0741d/yt_dlp/extractor/common.py#L1220-L1224

Reply:
default=None removes the warning

Reply:
I don't want to remove warning I just don't want `bug_reports_message()`:
```py
'please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U'
```
to be added.

Note: `e.msg` still has that added by the time it gets to be handled, so `e.orig_msg` is reported. (Sry, I didn't include that line.)

Reply:
Swings and roundabouts:
```diff
diff --git a/yt_dlp/extractor/safari.py b/yt_dlp/extractor/safari.py
index af32f9777..fcfc0159c 100644
--- a/yt_dlp/extractor/safari.py
+++ b/yt_dlp/extractor/safari.py
@@ -124,12 +124,14 @@ def _real_extract(self, url):
             if reference_id:
                 query['flashvars[referenceId]'] = reference_id
             else:
-                try:
-                    query['flashvars[referenceId]'] = self._search_regex(
-                        r'data-reference-id=(["\'])(?P<id>(?:(?!\1).)+)\1',
-                        webpage, 'kaltura reference id', group='id')
-                except RegexNotFoundError as e:
-                    self.report_warning(e.orig_msg)
+                _name = 'kaltura reference id'
+                reference_id = self._search_regex(
+                    r'data-reference-id=(["\'])(?P<id>(?:(?!\1).)+)\1',
+                    webpage, _name, default=None, group='id')
+                if reference_id is not None:
+                    query['flashvars[referenceId]'] = reference_id
+                else:
+                    self.report_warning('Unable to extract %s' % _name)
                     query['entry_id'] = self._search_regex(
                         r'data-entry-id=(["\'])(?P<id>(?:(?!\1).)+)\1',
                         webpage, 'kaltura entry id', group='id')
```

Reply:
What is the use of the warning if you don't want users to report it? Warnings should either be understandable to the user, or must let user know something is wrong in the code. `WARNING: Unable to extract kaltura reference id` accomplishes neither of these